src/app/manager-dashboard/certifications/certifications-add.component.ts

```diff
@@ -49,6 +51,9 @@ export class CertificationsAddComponent implements OnInit, AfterViewChecked {
           this.certificateForm.patchValue(certification);
           this.certificateInfo._rev = certification._rev;
           this.courseIds = certification.courseIds || [];
+          // here find out all courses in the system. compare all courses ids with this.coursesIds
+          const allCourseIds = []; // find out hhow to get values for this array.
+          this.enableAddCourses = deepEqual(allCourseIds, this.courseIds );
```

Comment:
There is an extra space before the last parenthesis that should be removed.

Reply:
I am having a problem with my code, where the value of "hasAllData" variable is always "undefined" and the "if"  statement does not get executed(line 62-65).
After debugging I noticed that 
1. this.courseTable does not have a dataTable.data property ( used this.courseTable.courses.data, instead)
2. The first time you click on a course and click the edit button, the this.courseTable.courses.data is undefined. If you return back to certificate list and click on the same course again, this.courseTable.courses.data has 23 entries.
3. The this.courseTable.courses.data must be local database data, feel like there should be more than 23 entries.
not sure how to go forward, what to try next to fix this issue. Any help would be appreciated.